packages/medusa/src/services/product-category.ts

```diff
@@ -98,20 +98,30 @@ class ProductCategoryService extends TransactionBaseService {
    * @return the product category.
    */
   async retrieve(
-    productCategoryId: string,
+    productCategoryIdOrHandle: string,
```

Comment:
thought: this is not a pattern we have used before as we generally only use ids for identification of items. I can't think of any direct bad consequences of it, but for the purpose of keeping things consistent, I lean toward not allowing it. Happy to hear @adrien2p or @olivermrbl on this too?

From the API perspective people should use the list endpoint to get a category with a specific handle:

`GET /product-categories?handle={handle}`

Reply:
Indeed it is not a common pattern in the code base. like for the product where we allow the handle query param to be passed we could do the same in the categories list end point as suggested by seb. Just for coherence purposes 👌alsoC since the handles are unique you are sure the list will return only one item

Reply:
All in for consistency, opened a new PR to make that happen - https://github.com/medusajs/medusa/pull/3825/files